### ISSUE TITLE
[airq] StringIndexOutOfBoundsException on empty usercalib

### ISSUE THREAD

**Issue by MikeTheTux** [openhab/openhab-addons#13928]
The binding throws a `StringIndexOutOfBoundsException`. I guess it happens in case of empty `"usercalib":{}`:

Trace:
```
...
2022-12-12 14:56:04.660 [DEBUG] [ab.binding.airq.internal.AirqHandler] - air-Q - airqHandler - getConfigData(): decObj={"mqtt":null, ... "usercalib":{}, ...}
...
2022-12-12 14:56:04.687 [TRACE] [ab.binding.airq.internal.AirqHandler] - air-Q - airqHandler - processType(): airqName=usercalib, channelName=userCalib, type=calib
2022-12-12 14:56:04.688 [WARN ] [ab.binding.airq.internal.AirqHandler] - air-Q - airqHandler - getConfigData(): Error in processConfigData(): java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 0
...
```

## Your Environment
openHAB 3.4.0.M4
My air-Q is actually quiet new. I did not perform any calibration. It is used without beeing connected to the cloud.

**Comment by MikeTheTux** [comment]
The same issue was already reported here: https://community.openhab.org/t/new-air-q-binding/115989/6

**Comment by MikeTheTux** [comment]
The following lenght-check in the java code of the binding fixes the issue:
```
if (str.length() >= 1) {
    updateState(channelName, new StringType(str.substring(0, str.length() - 1)));
} else {
    logger.trace(
            "air-Q - airqHandler - processType(): Cannot extract calibration data from this string: {}",
            dec.get(airqName).toString());
}
```

I'm thinking of creating a PR with those changes.

**Comment by lolodomo** [comment]
Rather `str.isEmpty()`

**Comment by MikeTheTux** [comment]
> Rather `str.isEmpty()`

In that case that would be the better choice, but there are other cases that need to check for a min lenght of 2:
```
if (arrstr.length() >= 2) {
    logger.trace("air-Q - airqHandler - processType(): property array {} set to {}",
            channelName, arrstr.substring(0, arrstr.length() - 2));
    getThing().setProperty(channelName, arrstr.substring(0, arrstr.length() - 2));
} else {
    logger.trace("air-Q - airqHandler - processType(): cannot handle this as an array: {}",
            proparr);
}
```

**Comment by lolodomo** [comment]
Closed by merged PR #14000
